doc/mgmt/OpenConfig_Interfaces.md

```diff
@@ -0,0 +1,199 @@
+    Configure/ Set Ethernet interface attributes.
+    Get Ethernet interface attributes.
+    Delete Ethernet interface attributes.
+3. Support port-speed and auto-negotiation on Ethernet interfaces via REST and gNMI.
+4. Support IPv4 and IPv6 address configuration on Ethernet interfaces via REST and gNMI.
+5. Support Enabling and disabling IPv6 enabled attribute on Ethernet interface via REST and gNMI. IPv6 enabled is set to disabled by default.
+### 1.1.2 Configuration and Management Requirements
+The Ethernet interface configurations can be done via REST and gNMI. The implementation will return an error if configuration is not allowed due to misconfiguration. There are no new configuration commands required to handle these configurations.
+### 1.1.3 Scalability Requirements
+The maximum number of interfaces depends on number of Ethernet interfaces supported by the switch.
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+SONiC already supports Ethernet interfaces configurations such as Get, Patch and Delete via REST and gNMI. This feature adds support for OpenConfig based YANG models using transformer based implementation instead of App.
+### 1.2.2 Container
+The code changes for this feature are part of *Management Framework* which includes the REST server and *gnmi* container for gNMI support in *sonic-mgmt-common* repository.
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+1. REST client through which the user can perform POST, PUT, PATCH, DELETE, GET operations on the supported YANG paths.
+2. gNMI client with support for capabilities get and set based on the supported YANG models.
+
+# 3 Design
+## 3.1 Overview
+This HLD design is in line with the [Management Framework HLD](https://github.com/project-arlo/SONiC/blob/354e75b44d4a37b37973a3a36b6f55141b4b9fdf/doc/mgmt/Management%20Framework.md)
+
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+There are no changes to CONFIG DB schema definition.
+### 3.2.2 APP DB
+There are no changes to APP DB schema definition.
+### 3.2.3 STATE DB
+There are no changes to STATE DB schema definition.
+### 3.2.4 ASIC DB
+There are no changes to ASIC DB schema definition.
+### 3.2.5 COUNTER DB
+There are no changes to COUNTER DB schema definition.
+
+## 3.3 CLI
+### 3.3.1 Data Models
+The higher speeds for Ethernet is brought in to openconfig-if-ethernet.yang from the [OpenConfig YANG](https://github.com/openconfig/public/blob/master/release/models/interfaces/openconfig-if-ethernet.yang#L263) 
+```
+    identity SPEED_200GB {
+    base ETHERNET_SPEED;
+    description "200 Gbps Ethernet";
+  }
+
+  identity SPEED_400GB {
+    base ETHERNET_SPEED;
+    description "400 Gbps Ethernet";
+  }
+
+  identity SPEED_600GB {
+    base ETHERNET_SPEED;
+    description "600 Gbps Ethernet";
+  }
+
+  identity SPEED_800GB {
+    base ETHERNET_SPEED;
+    description "800 Gbps Ethernet";
+  }
+  ```
+### 3.3.2 REST API Support
+#### 3.3.2.1 GET
+Supported
+```

```

Comment:
Also we can share sample o/p of leaf level GET, PATCH for one attribute as e.g. to show what will be supported.